presto-spi/src/main/java/com/facebook/presto/spi/block/BlockFlattener.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import com.facebook.presto.spi.api.Experimental;
+
+import java.util.Arrays;
+
+import static com.facebook.presto.spi.block.ClosingBlockLease.newLease;
+import static java.util.Objects.requireNonNull;
+
+@Experimental
+public class BlockFlattener
```

Comment:
Is there any particular reason to have this class in the SPI? Can it go into presto-main?

Reply:
It relies on package-private methods for the types it flattens.